samples/net/cellular_modem/src/main.c

```diff
@@ -419,25 +447,35 @@ int main(void)
 
 	init_sample_test_packet();
 
+	ppp_iface = net_if_get_first_by_type(&NET_L2_GET_NAME(PPP));
+	net_mgmt_init_event_callback(&l4_cb, l4_event_handler, L4_EVENT_MASK);
+	net_mgmt_add_event_callback(&l4_cb);
+
 	printk("Powering on modem\n");
 	pm_device_action_run(modem, PM_DEVICE_ACTION_RESUME);
 
 	printk("Bring up network interface\n");
-	ret = net_if_up(iface);
+	ret = net_if_up(ppp_iface);
 	if (ret < 0) {
 		printk("Failed to bring up network interface\n");
 		return -1;
 	}
 
-	printk("Waiting for L4 connected & DNS server added\n");
-	ret = net_mgmt_event_wait_on_iface(iface, NET_EVENT_L4_CONNECTED | NET_EVENT_DNS_SERVER_ADD,
-					   NULL, NULL, NULL, K_SECONDS(120));
+	printk("Waiting for L4 connected\n");
+	ret = k_event_wait(&l4_event, L4_CONNECTED, false, K_SECONDS(120));
 
-	if (ret != 0) {
+	if (!ret ) {
```

Comment:
Better to have `if (ret == 0) {` as we try to avoid using boolean operations for non boolean variables.